### ISSUE TITLE
Analyze changes to CloudTrail events published by AWS Secrets Manager

### ISSUE THREAD

**Issue by Selutario** [wazuh/wazuh#21648]
## Description

As mentioned [here](https://kb.wisc.edu/news.php?id=13232), in January 2024, AWS Secrets Manager has published a change to the format of the secret ARN response element in the CloudTrail event for some of the AWS Secrets Manager APIs:

> Currently, 10 events, listed at the end of this email, are published with “aRN” as the response element while the rest of the events are published with “arn” as the response element. If you are consuming any of these 10 AWS CloudTrail Events in your workloads either directly or using Amazon EventBridge, we recommend you update your code to account for both formats of the secret ARN response element (“arn” and “aRN”) to ensure your code does not break with this upcoming change. After January 2024, you can choose to either keep the updated code to support both formats or change it to consume only “arn” format.

We need to review if this affects our integration with Cloudtrail or its rules, and if so, apply the appropriate changes to keep everything working.

## Checks

The following elements have been updated or reviewed (should also be checked if no modification is required):
- [x] Tests (unit tests, API integration tests).
- [x] Changelog.
- [x] Documentation.
- [x] Integration test mapping (using `api/test/integration/mapping/_test_mapping.py`).

**Comment by javiersanchz** [comment]
## Update 

The `CloudTrail` logic regarding the new version of logs was reviewed, which does not seem to affect these at any point.

The set of rules related to `aws_cloudtrail` was also examined, and it has no direct references to the "`arn`" field that could be affected by the format change.

The decoders related to AWS were also checked, and nothing related to these changes was found. The only difference found is in `/ruleset/decoders/0565-aws-eks-authenticator_decoders.xml`:

https://github.com/wazuh/wazuh/blob/f4d39045b00e54cbf725048004c9e2efa4bbbaa2/ruleset/decoders/0565-aws-eks-authenticator_decoders.xml#L30-L34

This decoder is designed to specifically work with fields following the "arn" format, which is already aligned with the new format.

**In summary, based on the reviewed code, rules, and decoders, you should not encounter issues related to the transition to "`arn`"**

- As a result, the PR that was created at https://github.com/wazuh/wazuh/pull/21700 has been closed. This PR included changes to the function responsible for checking the input parameter args_valid_iam_role_arn. However, it is not directly related to the new changes described in the issue

**Comment by fdalmaup** [comment]
# Review

The analysis was done correctly, checking the code and the default Wazuh rules and decoders related to the AWS module to check the impact of the standardization of the field.

LGTM!

**Comment by Selutario** [comment]
Understood, thanks @javiersanchz @fdalmaup :+1:
